packages/backend-function/src/function_client_config_generator.ts

```diff
@@ -0,0 +1,128 @@
+import fs from 'fs';
+import { EOL } from 'os';
+import path from 'path';
+
+const lambdaConfigHelper = (
+  functionName: string
+) => `import { env } from "../env/${functionName}";
+import modelIntrospection from "../model-introspection-schema/${functionName}.json";
+
+import {
+  CredentialsAndIdentityIdProvider,
+  CredentialsAndIdentityId,
+} from "aws-amplify/auth";
+
+class CustomCredentialsProvider implements CredentialsAndIdentityIdProvider {
+  env: Env;
+  constructor(env: Env) {
+    this.env = env;
+  }
+
+  async getCredentialsAndIdentityId(): Promise<
+    CredentialsAndIdentityId | undefined
+  > {
+    const credentials: CredentialsAndIdentityId = {
+      credentials: {
+        accessKeyId: this.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: this.env.AWS_SECRET_ACCESS_KEY,
+        sessionToken: this.env.AWS_SESSION_TOKEN,
+      },
+    };
+    return credentials;
+  }
+  clearCredentialsAndIdentityId(): void {}
+}
+
+type Env = {
+  AWS_ACCESS_KEY_ID: string;
+  AWS_SECRET_ACCESS_KEY: string;
+  AWS_SESSION_TOKEN: string;
+  AMPLIFY_DATA_GRAPHQL_ENDPOINT?: string;
+  AWS_REGION: string;
+};
+
+/**
+ * Configures the amplify library for lambda execution with data-schema client configuration
+ *
+ * @param env The env vars provided for lambda execution
+ * @param modelIntrospection The MIS generated for the schema
+ * @returns the two named inputs to use with \`Amplify.configure(resourceConfig, libraryOptions)\`
+ */
+function lambdaConfig(env: Env, modelIntrospection: object) {
+  const customCredentialsProvider = new CustomCredentialsProvider(env);
+
+  return {
+    resourceConfig: {
+      API: {
+        GraphQL: {
+          endpoint: env.AMPLIFY_DATA_GRAPHQL_ENDPOINT ?? "",
+          region: env.AWS_REGION,
+          defaultAuthMode: "iam" as const,
+          modelIntrospection: modelIntrospection as any,
+        },
+      },
+    },
+    libraryOptions: {
+      Auth: {
+        credentialsProvider: customCredentialsProvider,
+      },
+    },
+  };
+};
+const { resourceConfig, libraryOptions } = lambdaConfig(
+  env,
+  modelIntrospection
+);
+
+export { resourceConfig, libraryOptions };
+`;
+
+/**
+ * Generates a client config for environment variables
+ */
+export class FunctionClientConfigGenerator {
+  private readonly header =
+    '// This file is auto-generated by Amplify. Edits will be overwritten.';
+
+  private clientConfigFilePath: string;
+
+  private indentation: string = '  ';
+
+  /**
+   * Initialize client config file name and location
+   */
+  constructor(private readonly functionName: string) {
+    this.clientConfigFilePath = `${process.cwd()}/.amplify/generated/client-config/${
+      this.functionName
+    }.ts`;
+  }
+  /**
+   * Generate a client configuration
+   */
+  generateClientConfig = () => {
+    this.writeFile(
+      [this.header, lambdaConfigHelper(this.functionName)].join(EOL)
```

Comment:
https://www.google.com/search?q=helper+code+smell

can this be `generateAmplifyClientInitializationCode` ??